src/Sparrow/Platform/Posix/SysUtils.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using Sparrow.Logging;
+using Voron.Platform.Posix;
+
+namespace Sparrow.Platform.Posix
+{
+    public static class SysUtils
+    {
+        private static Logger _logger = LoggingSource.Instance.GetLogger("SysUtils", "Raven/Server");
+
+        public static unsafe ulong ReadULongFromFile(string filename)
+        {
+            var fd = Syscall.open(filename, OpenFlags.O_RDONLY, FilePermissions.S_IRUSR);
+            if (fd < 0)
+            {
+                if (_logger.IsInfoEnabled)
+                    _logger.Info($"Cannot open '{filename}'. Will not respect container's memory limit if running under one.");
+                return ulong.MaxValue;
+            }
+
+            ulong cgroup;
+            UIntPtr readSize = (UIntPtr)32;
+            IntPtr pBuf = Marshal.AllocHGlobal((int)readSize);
+            try
+            {
+                Memory.Set((byte*)pBuf, 0, 32);
+                var cgroupRead = Syscall.read(fd, pBuf.ToPointer(), (ulong)readSize);
+                if (cgroupRead > 30 || cgroupRead == 0) // check we are not garbadged
+                {
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"Got invalid number of characters ({cgroupRead}) from filename. Will not respect container's memory limit if running under one.");
+                    Syscall.close(fd);
+                    return ulong.MaxValue;
+                }
+
+                Syscall.close(fd);
+
+                string str = null;
+                try
+                {
+                    str = Encoding.ASCII.GetString((byte*)pBuf.ToPointer(), (int)cgroupRead);
+                    cgroup = Convert.ToUInt64(str);
```

Comment:
This allocates a string, just to get the ulong for it.
Looks like a waste. Either we parse it ourselves and do no memory allocation, or we just use File.ReadAllText() and save all this code.